Reorg cache code to make it simpler

aorenste opened the pull request:
Summary:
Pull the big nested function out of the middle of cached_autotune() into its own class.

Also refactor creating the autotune cache itself out - which gets shared in the next diff.

Test Plan: unit tests

Differential Revision: D60677501


cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang

oulgen reviewed (approved):
comments left on the internal diff

aorenste committed:
Reorg cache code to make it simpler (#134911)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/134911

Pull the big nested function out of the middle of cached_autotune() into its own class.

Also refactor creating the autotune cache itself out - which gets shared in the next diff.

Test Plan: unit tests

Reviewed By: oulgen

Differential Revision: D60677501
